staticaddr/loopin/manager.go

```diff
@@ -189,12 +206,217 @@ func (m *Manager) Run(ctx context.Context, currentHeight uint32) error {
+	// find them in the requested sweep.
+	depositToIdxMap := make(map[string]int)
+	for reqOutpoint := range req.DepositToNonces {
+		hasDeposit := false
+		for _, depositOutpoint := range loopIn.DepositOutpoints {
+			if depositOutpoint == reqOutpoint {
+				hasDeposit = true
+				break
+			}
+		}
+		if !hasDeposit {
+			return fmt.Errorf("deposit outpoint not part of " +
+				"loop-in")
+		}
+
+		foundDepositInTx := false
+
+		for i, txIn := range sweepTx.TxIn {
+			if txIn.PreviousOutPoint.String() == reqOutpoint {
+				depositToIdxMap[reqOutpoint] = i
+				foundDepositInTx = true
+				break
+			}
+		}
+
+		if !foundDepositInTx {
+			return fmt.Errorf("deposit outpoint not part of " +
+				"sweep tx")
+		}
+	}
+
+	prevoutMap := make(map[wire.OutPoint]*wire.TxOut)
+
+	// Set all the prevouts in the prevout map.
+	for i := range req.PrevoutInfo {
+		prevout := req.PrevoutInfo[i]
+
+		txid, err := chainhash.NewHash(prevout.TxidBytes)
+		if err != nil {
+			return err
+		}
+
+		prevoutMap[wire.OutPoint{
+			Hash:  *txid,
+			Index: prevout.OutputIndex,
+		}] = &wire.TxOut{
+			Value:    int64(req.PrevoutInfo[i].Value),
+			PkScript: req.PrevoutInfo[i].PkScript,
+		}
+	}
+
+	prevOutputFetcher := txscript.NewMultiPrevOutFetcher(
+		prevoutMap,
+	)
+
+	sigHashes := txscript.NewTxSigHashes(
+		sweepPacket.UnsignedTx, prevOutputFetcher,
+	)
+
+	// We'll now sign for every deposit that is part of the loop-in.
+	responseMap := make(map[string]*looprpc.ClientSweeplessSigningInfo)
+	for depositOutpoint, nonce := range req.DepositToNonces {
+		taprootSigHash, err := txscript.CalcTaprootSignatureHash(
+			sigHashes, txscript.SigHashDefault, sweepPacket.UnsignedTx,
+			depositToIdxMap[depositOutpoint], prevOutputFetcher,
+		)
+		if err != nil {
+			return err
+		}
+
+		var (
+			serverNonce [musig2.PubNonceSize]byte
+			sigHash     [32]byte
+		)
+
+		copy(serverNonce[:], nonce)
+		musig2Session, err := loopIn.createMusig2Session(
+			ctx, m.cfg.Signer,
+		)
+		if err != nil {
+			return err
+		}
+
+		haveAllNonces, err := m.cfg.Signer.MuSig2RegisterNonces(
+			ctx, musig2Session.SessionID,
+			[][musig2.PubNonceSize]byte{serverNonce},
+		)
+		if err != nil {
+			return err
+		}
+
+		if !haveAllNonces {
+			return fmt.Errorf("expected all nonces to be " +
+				"registered")
+		}
+
+		copy(sigHash[:], taprootSigHash)
+
+		// Since our MuSig2 session has all nonces, we can now create
+		// the local partial signature by signing the sig hash.
+		sig, err := m.cfg.Signer.MuSig2Sign(
+			ctx, musig2Session.SessionID, sigHash, true,
+		)
+		if err != nil {
+			return err
+		}
+
+		responseMap[depositOutpoint] = &looprpc.ClientSweeplessSigningInfo{ //nolint:lll
+			Nonce: musig2Session.PublicNonce[:],
+			Sig:   sig,
+		}
+	}
+
+	txHash := sweepTx.TxHash()
+
+	_, err = m.cfg.Server.PushStaticAddressSweeplessSigs(
+		ctx, &looprpc.PushStaticAddressSweeplessSigsRequest{
+			SwapHash:    loopIn.SwapHash[:],
+			Txid:        txHash[:],
```

Comment:
IIUC, server knows txhash already, because it depends on unsigned parts only. Can we remove the field?